src/backends/odbc/standard-use-type.cpp

```diff
@@ -23,50 +24,70 @@ void* odbc_standard_use_type_backend::prepare_for_bind(
     switch (type_)
     {
     // simple cases
-    case x_short:
+    case x_int8:
+        sqlType = SQL_TINYINT;
+        cType = SQL_C_STINYINT;
+        size = sizeof(int8_t);
+        break;
+    case x_uint8:
+        sqlType = SQL_TINYINT;
+        cType = SQL_C_UTINYINT;
+        size = sizeof(uint8_t);
+        break;
+    case x_int16:
         sqlType = SQL_SMALLINT;
         cType = SQL_C_SSHORT;
-        size = sizeof(short);
+        size = sizeof(int16_t);
+        break;
+    case x_uint16:
+        sqlType = SQL_SMALLINT;
+        cType = SQL_C_USHORT;
+        size = sizeof(uint16_t);
         break;
-    case x_integer:
+    case x_int32:
         sqlType = SQL_INTEGER;
         cType = SQL_C_SLONG;
-        size = sizeof(int);
+        size = sizeof(int32_t);
+        break;
+    case x_uint32:
+        sqlType = SQL_INTEGER;
+        cType = SQL_C_ULONG;
+        size = sizeof(uint32_t);
         break;
-    case x_long_long:
+    case x_int64:
         if (use_string_for_bigint())
         {
           sqlType = SQL_NUMERIC;
           cType = SQL_C_CHAR;
           size = max_bigint_length;
           buf_ = new char[size];
           snprintf(buf_, size, "%" LL_FMT_FLAGS "d",
-                   exchange_type_cast<x_long_long>(data_));
+                   static_cast<long long>(exchange_type_cast<x_int64>(data_)));
```

Comment:
I think the casts here (and below) are unnecessary as well.

Reply:
This is the one of the only places where the cast is necessary. `int64_t` isn't defined as `long long` on every platform. The format specifier `long long` is used as this should be the common ground for 64-bit integers across all platforms. If we don't cast here, the compiler complains with a warning.